### DISCUSSION TITLE
DiodeModel Cloning doesn't work?

### DISCUSSION THREAD

**Discussion by DeadTomGC** [SpiceSharp/SpiceSharpBehavioral#44] [Q&A] [answered]
I was trying to clone a circuit that includes a Diode model.  However, I couldn't.   I therefore just tried making and cloning a single DiodeModel, and couldn't.  Any idea what's going on here?  Sorry for the frequency of posts, but I am doing a lot of experimental work with this library at the moment.

`circuit.Add(new DiodeModel("DN4xx"));
var clone = circuit["DB4xx"].Clone();'

This gave an error from the 'SetValue' Function saying that the value must be less than 0.  The value in question was '{-1 {not set}}' Which... seems less than 0 to me.

This is really one of those, "I must be an idiot" moments.

Sorry that this may not be exact code, but the company I work for isn't too fond of code being copied out to other places, so I'm just recreating the relevant bits.  I can get more precise with my transcription of the code if necessary.

**Comment by svenboulanger** [comment, marked as answer]
Hi! This is indeed an issue. It turns out that the checking function "LessThan" appears only once in the whole project: for the breakdown voltage of a diode model.
But unfortunately this function is implemented incorrectly and will throw errors for correct values. I'll create another issue for that.
You keep finding these things :-) Good work!
